src/Microsoft.ML.Transforms/Text/TextNormalizerTransform.cs

```diff
@@ -92,294 +92,498 @@ private static VersionInfo GetVersionInfo()
+        /// <param name="textCase">Casing text using the rules of the invariant culture.</param>
+        /// <param name="keepDiacritics">Whether to keep diacritical marks or remove them.</param>
+        /// <param name="keepPunctuations">Whether to keep punctuation marks or remove them.</param>
+        /// <param name="keepNumbers">Whether to keep numbers or remove them.</param>
+        public TextNormalizerEstimator(IHostEnvironment env,
+            string inputColumn,
+            string outputColumn = null,
+            CaseNormalizationMode textCase = Defaults.TextCase,
+            bool keepDiacritics = Defaults.KeepDiacritics,
+            bool keepPunctuations = Defaults.KeepPunctuations,
+            bool keepNumbers = Defaults.KeepNumbers)
+            : this(env, new TextNormalizerTransform.ColumnInfo(inputColumn, outputColumn ?? inputColumn, textCase, keepDiacritics, keepPunctuations, keepNumbers))
+        {
+        }
+
+        /// <summary>
+        /// Normalizes incoming text in input columns by changing case, removing diacritical marks, punctuation marks and/or numbers
+        /// and outputs new text as output columns.
+        /// </summary>
+        /// <param name="env">The environment.</param>
+        /// <param name="columns">Pairs of columns to run the text normalization on.</param>
+        /// <param name="textCase">Casing text using the rules of the invariant culture.</param>
+        /// <param name="keepDiacritics">Whether to keep diacritical marks or remove them.</param>
+        /// <param name="keepPunctuations">Whether to keep punctuation marks or remove them.</param>
+        /// <param name="keepNumbers">Whether to keep numbers or remove them.</param>
+        public TextNormalizerEstimator(IHostEnvironment env,
+            (string input, string output)[] columns,
+            CaseNormalizationMode textCase = Defaults.TextCase,
+            bool keepDiacritics = Defaults.KeepDiacritics,
+            bool keepPunctuations = Defaults.KeepPunctuations,
+            bool keepNumbers = Defaults.KeepNumbers)
+            : this(env, columns.Select(x => new TextNormalizerTransform.ColumnInfo(x.input, x.output, textCase, keepDiacritics, keepPunctuations, keepNumbers)).ToArray())
+        {
+        }
 
-            // Basic combining diacritics
-            return ch >= 0x0300 && ch <= 0x036F ||
+        public TextNormalizerEstimator(IHostEnvironment env, params TextNormalizerTransform.ColumnInfo[] columns)
+          : this(env, new TextNormalizerTransform(env, columns))
+        {
+        }
 
-                // Hebrew combining diacritics
-                ch >= 0x0591 && ch <= 0x05BD || ch == 0x05C1 || ch == 0x05C2 || ch == 0x05C4 ||
-                ch == 0x05C5 || ch == 0x05C7 ||
+        private TextNormalizerEstimator(IHostEnvironment env, TextNormalizerTransform transformer)
```

Comment:
>TextNormalizerEstimator [](start = 16, length = 23)

When do we use this constructor? The above TextNormalizerEstimator(IHostEnvironment env, params TextNormalizerTransform.ColumnInfo[] columns) can call directly into base.

Reply:
in public constructor above

---
In reply to: [226028712](https://github.com/dotnet/machinelearning/pull/1276#discussion_r226028712) [](ancestors = 226028712)

Reply:
How about: 

public TextNormalizerEstimator(IHostEnvironment env, params TextNormalizerTransform.ColumnInfo[] columns)
          :base(Contracts.CheckRef(env, nameof(env)).Register(nameof(TextNormalizerEstimator)), new TextNormalizerTransform(env, columns))
        {
        }

And we get rid of the private one? 

---
In reply to: [226032246](https://github.com/dotnet/machinelearning/pull/1276#discussion_r226032246) [](ancestors = 226032246,226028712)